src/getPage.tsx

```diff
@@ -79,7 +82,10 @@ export default async function getPage({
       <RouterProvider pageObject={pageObject} options={options}>
         <NavigationProvider
           makePage={async (route) => {
-            const { pageElement } = await makePage(route);
+            const { pageElement } = await makePage(route, (req) => {
+              req.headers.cookie = document.cookie;
```

Comment:
On client side navigation we are just passing forward the `document.cookie`. This is where we could also pass the `referer` header (this is how browser would behave).